bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerServlet.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2017 Adobe.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.adobe.acs.commons.mcp.impl;
+
+import com.adobe.acs.commons.mcp.ControlledProcessManager;
+import com.adobe.acs.commons.mcp.ProcessDefinition;
+import com.adobe.acs.commons.mcp.ProcessInstance;
+import com.adobe.acs.commons.mcp.util.DeserializeException;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.jcr.RepositoryException;
+import javax.servlet.ServletException;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.sling.SlingServlet;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Servlet for interacting with MCP.
+ */
+@SlingServlet(paths = "/bin/mcp")
```

Comment:
Typically we attach the POST behavior to the resource that is exposing this UI view via a selector (or occasionally sub-resource). So if this tool is exposed via say `/etc/acs-commons/mcp` ... `HTTP POST` to `/etc/acs-commons/mcp/jcr:content.process.json` (and that  `jcr:content` has a `'sling:resourceType` that maps to this servlet + selector). 

Since we have to bring the resource to expose the WebUI, it makes sense to use that for other interactions w the tool as well.